capstone/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -1,11 +1,269 @@
+      String loginUrl = userService.createLoginURL(
+        URL_TO_REDIRECT_AFTER_USER_LOGS_IN
+      );
+      userDetails.put("loginUrl", loginUrl);
+      userDetails.put("status_code", "401");
+      return userDetails;
+    }
+    return checkForStudentEmail(
+      pool,
+      userService,
+      get_user_row_query,
+      response
+    );
+  }
+
+  /**
+   * This method checks if the email supplied is a student email by confirming the '.edu' tld.
+   * @param pool It is a DataSource object that serves to interact with our database connection.
+   * @param userService This is a UserService object that holds information about the current user.
+   * @param get_user_row_query This is an SQL statement to query our database for the needed user's information.
+   * @param response A HttpServletResponse object that holds the information that will be sent back to the user.
+   * @return HashMap<String, String> saves a key-value pair of all neccessary user details derived.
+   */
+  public HashMap<String, String> checkForStudentEmail(
+    DataSource pool,
+    UserService userService,
+    String get_user_row_query,
+    HttpServletResponse response
+  )
+    throws SQLException, IOException {
+    HashMap<String, String> userDetails = new HashMap<>();
+    String email = userService.getCurrentUser().getEmail();
+    if (!email.endsWith(".edu")) {
+      userDetails.put("status_code", "412");
+      String logoutUrl = userService.createLogoutURL(
+        URL_TO_REDIRECT_AFTER_USER_LOGS_OUT
+      );
+      userDetails.put("logoutUrl", logoutUrl);
+      return userDetails;
+    }
+    return checkIfUserIsAlreadyRegistered(
+      pool,
+      userService,
+      get_user_row_query,
+      response
+    );
+  }
+
+  /**
+   * This method checks if the current user is already in  our database, if not we redirect them to fill in their info.
+   * @param pool It is a DataSource object that serves to interact with our database connection.
+   * @param userService This is a UserService object that holds information about the current user.
+   * @param get_user_row_query This is an SQL statement to query our database for the needed user's information.
+   * @param response A HttpServletResponse object that holds the information that will be sent back to the user.
+   * @return HashMap<String, String> saves a key-value pair of all neccessary user details derived.
+   */
+  public HashMap<String, String> checkIfUserIsAlreadyRegistered(
+    DataSource pool,
+    UserService userService,
+    String get_user_row_query,
+    HttpServletResponse response
+  )
+    throws SQLException, IOException {
+    HashMap<String, String> userDetails = new HashMap<>();
+    String email = userService.getCurrentUser().getEmail();
+    Boolean isUserLoggedIn = userService.isUserLoggedIn();
+    try (Connection conn = pool.getConnection()) {
+      try (
+        PreparedStatement queryStatement = conn.prepareStatement(
+          get_user_row_query
+        )
```

Comment:
combine this with the try above:
try(Connection conn = pool.getConnection();
      PreparedStatement queryStatement = conn.prepareStatement(
          get_user_row_query) {
}